packages/react-client/src/hooks/tsClient.ts

```diff
@@ -0,0 +1,211 @@
+import { useCallback, useMemo, useRef, useSyncExternalStore } from "react";
+import type {
+  BandwidthLimit,
+  Component,
+  Endpoint,
+  MessageEvents,
+  Peer,
+  SimulcastConfig,
+  TrackBandwidthLimit,
+  TrackContext,
+  Encoding,
+} from "@fishjam-cloud/ts-client";
+import type { PeerMetadata, TrackMetadata, ConnectConfig } from "../types";
+import type { PeerState, Track, TrackId } from "../state.types";
+import { useFishjamContext } from "./fishjamContext";
+
+const eventNames = [
+  "socketClose",
+  "socketError",
+  "socketOpen",
+  "authSuccess",
+  "authError",
+  "disconnected",
+  "reconnectionStarted",
+  "reconnected",
+  "reconnectionRetriesLimitReached",
+  "joined",
+  "joinError",
+  "trackReady",
+  "trackAdded",
+  "trackRemoved",
+  "trackUpdated",
+  "peerJoined",
+  "peerLeft",
+  "peerUpdated",
+  "componentAdded",
+  "componentRemoved",
+  "componentUpdated",
+  "connectionError",
+  "tracksPriorityChanged",
+  "bandwidthEstimationChanged",
+  "targetTrackEncodingRequested",
+  "localTrackAdded",
+  "localTrackRemoved",
+  "localTrackReplaced",
+  "localTrackMuted",
+  "localTrackUnmuted",
+  "localTrackBandwidthSet",
+  "localTrackEncodingBandwidthSet",
+  "localTrackEncodingEnabled",
+  "localTrackEncodingDisabled",
+  "localPeerMetadataChanged",
+  "localTrackMetadataChanged",
+  "disconnectRequested",
+] as const satisfies (keyof MessageEvents<unknown, unknown>)[];
+
+interface FishjamClientState {
+  peers: PeerState[];
+  components: PeerState[];
+  localPeer: PeerState | null;
+  isReconnecting: boolean;
+}
+
+function trackContextToTrack(track: TrackContext<PeerMetadata, TrackMetadata>): Track {
+  return {
+    rawMetadata: track.rawMetadata,
+    metadata: track.metadata,
+    trackId: track.trackId,
+    stream: track.stream,
+    simulcastConfig: track.simulcastConfig ?? null,
+    encoding: track.encoding ?? null,
+    vadStatus: track.vadStatus,
+    track: track.track,
+    metadataParsingError: track.metadataParsingError,
+  };
+}
+
+function endpointToPeerState(
+  peer:
+    | Peer<PeerMetadata, TrackMetadata>
+    | Component<PeerMetadata, TrackMetadata>
+    | Endpoint<PeerMetadata, TrackMetadata>,
+): PeerState {
+  const tracks = [...peer.tracks].reduce(
+    (acc, [, track]) => ({ ...acc, [track.trackId]: trackContextToTrack(track) }),
+    {} as Record<TrackId, Track>,
+  );
+  return {
+    rawMetadata: peer.rawMetadata,
+    metadata: peer.metadata,
+    metadataParsingError: peer.metadataParsingError,
+    id: peer.id,
+    tracks,
+  };
+}
+
+export const useFishjamClient = () => {
```

Comment:
Let's remove this export from API in future ticket, and for now it could be marked as deprecated
```suggestion
/**
 * @deprecated WIll be removed from public API in  FCE-XXX
 */
export const useFishjamClient = () => {
```